### PR TITLE
[Feature] (ODBC) support read/write emoji of utf16 via odbc table

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #11728 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/docs/ecosystem/external-table/odbc-of-doris.md [hunk 1]
```diff
@@ -371,3 +371,7 @@ sudo alien -i  oracle-instantclient19.13-sqlplus-19.13.0.0.0-2.x86_64.rpm
 10. 报错`driver connect Err: xxx`
 
     通常是连接数据库失败，Err部分代表了不同的数据库连接失败的报错。这种情况通常是配置存在问题。可以检查是否错配了ip地址，端口或账号密码。
+    
+11. 读写mysql外表的emoji表情出现乱码
+
+    Doris进行odbc外表连接时，默认采用的编码为utf8，由于mysql之中默认的utf8编码为utf8mb3，无法表示需要4字节编码的emoji表情。这里需要在建立mysql外表时设置`charset`=`utf8mb4`，便可以正常读写emoji表情😀。
```
Comment (morningman) [inline docs/zh-CN/docs/ecosystem/external-table/odbc-of-doris.md:377]:
```suggestion
    Doris进行odbc外表连接时，默认采用的编码为utf8，由于mysql之中默认的utf8编码为utf8mb4，无法表示需要4字节编码的emoji表情。这里需要在建立mysql外表时设置`charset`=`utf8mb4`，便可以正常读写emoji表情😀。
```

File: build.sh [hunk 2]
```diff
@@ -120,7 +120,7 @@ fi
 
 eval set -- "$OPTS"
 
-PARALLEL=$[$(nproc)/4+1]
+PARALLEL=$[$(nproc)/2+1]
```
Comment (morningman) [inline build.sh:123]:
```suggestion
PARALLEL=$[$(nproc)/4+1]
```

File: docs/en/docs/ecosystem/external-table/odbc-of-doris.md [hunk 3]
```diff
@@ -381,5 +381,8 @@ This is the compatibility problem between MySQL database ODBC driver and existin
 
     Connection to the database fails. The` Err: part` represents the error of different database connection failures. This is usually a configuration problem. You should check whether the IP address, port or account password are mismatched.
 
-    
+ 11. Messy code appears when reading and writing emoji emoji in mysql odbc table
+
+    The default encoding used by Doris when connecting to odbc tables is utf8, since the default utf8 encoding in mysql is utf8mb3, it can't represent the emoji expressions which need 4-byte encoding. Here need to set `charset`=`utf8mb4` when you create odbc mysql tables, then can read and write emoji emojis normally 😀.
```
Comment (jacktengg) [inline docs/en/docs/ecosystem/external-table/odbc-of-doris.md:386]: 'emoji emojis' typos?

**Comment by jacktengg** [comment]
tried with mysql server 8.0, it's OK

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**